shared/src/lib.rs

```diff
@@ -19,8 +19,10 @@ pub struct SendApiKeyMessage {
     pub request_body: String,
     /// The HTTP verb to use
     pub http_verb: String,
-    /// The full URL for the HTTP request
-    pub api_url: String,
+    /// The URL base for the HTTP request ex:(http://127.0.0.1:3002")
+    pub api_url_base: String,
```

Comment:
I dont think the client should split the url into two parts, that can happen on the back end. As in i would keep the full url here and split it out in the `make_request` request handler.  This also works though.